### PR TITLE
Fix bug to overwrite all subsections definition by subclass

### PR DESCRIPTION
**Pull request by tagomoris**
* overwriting config_section options like required/multi/alias breaks behavior of `super`
  * so subclass must not overwrite these options
* subsection definitions of subsections (Hash) is simply merged
  * it is bug, because same section definition like <a> is overwritten by subclass
  * under this bug, subclass MUST repeat superclass's definition, and add original definition
  * but this is impossible in the real world
  * subsection definitions should be merged correctly

This bug especially make effect to the situations such like `<buffer>` configuration sections.
* Once Buffer base class define `<buffer>` section as non-required and non-multi
* Next AwesomeBuffer class (subclass of base class) define `<buffer>` section as required and multi
* `super` of AwesomeBuffer must raises error always

With automatically merged `config_section`, such that case doesn't occur.

### CODE REVIEW

File: test/config/test_configurable.rb [hunk 1]
```diff
@@ -493,6 +499,59 @@ def e(name, arg = '', attrs = {}, elements = [])
...
+          # <detail> address </detail> is required by Example1 config_param spec
+          assert_raise(Fluent::ConfigError.new("'address' parameter is required, in section detail")) { ConfigurableSpec::Example2.new.configure(conf2) }
+
+          conf3 = e('ROOT', '', {
+              'name' => 'tagomoris',
+              'bool' => true,
+            },
+            [e('detail', '', { 'address' => "Chiyoda Tokyo Japan" }, [])],
+          )
+          # <detail> phone_no </detail is required by Example2 config_param spec
```
Comment (repeatedly) [inline test/config/test_configurable.rb]: `</detail` lack of `>`

Comment (tagomoris) [inline test/config/test_configurable.rb]: Ah, fixing...

**Comment by tagomoris** [comment]
typo fixed.

File: lib/fluent/config/configure_proxy.rb [hunk 2]
```diff
@@ -148,11 +161,16 @@ def config_section(name, *args, &block)
           raise ArgumentError, "#{self.name}: unknown config_section arguments: #{args.inspect}"
         end
 
-        sub_proxy = ConfigureProxy.new(name, *args)
-        sub_proxy.instance_exec(&block)
+        # if @sections[name]
```
Comment (repeatedly) [inline lib/fluent/config/configure_proxy.rb]: Why these lines are commented out?
Can't be deleted?

Comment (tagomoris) [inline lib/fluent/config/configure_proxy.rb]: Ah, it's remains of try-and-errors. I'll delete it.

**Comment by tagomoris** [comment]
fixed and force pushed to remove remains of try&errors.

**Comment by sonots** [comment]
cherry-picked to v0.10

**Pull request merged**